python/ray/serve/client.py

```diff
@@ -120,6 +120,12 @@ def shutdown(self) -> None:
         Shuts down all processes and deletes all state associated with the
         instance.
         """
+
+        # shutdown handles
+        for k in list(self.handle_cache):
+            self.handle_cache[k].stop_metrics_pusher()
+            del self.handle_cache[k]
```

Comment:
do we need explicit del here?

Reply:
This is for optimization. I want the handle reference goes to 0 as soon as possible. And the background thread is going to be dead and controller is going to be shut down. I think we are good  to removed from the cache side.